contracts/StablecoinConverter.sol

```diff
@@ -101,79 +154,59 @@ contract StablecoinConverter is EpochTokenLocker {
-            delete orders[msg.sender][id[i]];
+    /** @dev A user facing function used to delete expired orders.
+      * This release of storage gives a gas refund to msg.sender and requires that all orders are expired.
+      * @param ids referencing the indices of user's orders to be deleted
+      *
+      * Requirements:
+      * - Each requested order is expired
+      */
+    function freeStorageOfOrder(uint[] memory ids) public {
+        for (uint i = 0; i < ids.length; i++) {
+            require(orders[msg.sender][ids[i]].validUntil + 1 < getCurrentBatchId(), "Order is still valid");
+            delete orders[msg.sender][ids[i]];
         }
     }
 
-    function tokenAddressToIdMap(address addr) public view returns (uint16) {
-        return IdToAddressBiMap.getId(registeredTokens, addr);
-    }
-
-    function tokenIdToAddressMap(uint16 id) public view returns (address) {
-        return IdToAddressBiMap.getAddressAt(registeredTokens, id);
-    }
-
-    function getEncodedAuctionElements() public view returns (bytes memory elements) {
-        if (allUsers.size() > 0) {
-            address user = allUsers.first();
-            bool stop = false;
-            while (!stop) {
-                for (uint i = 0; i < orders[user].length; i++) {
-                    Order memory order = orders[user][i];
-                    elements = elements.concat(encodeAuctionElement(
-                        user,
-                        getBalance(user, tokenIdToAddressMap(order.sellToken)),
-                        order
-                    ));
-                }
-                if (user == allUsers.last) {
-                    stop = true;
-                } else {
-                    user = allUsers.next(user);
-                }
-            }
-        }
-        return elements;
-    }
-
-    mapping (uint16 => uint128) public currentPrices;
-    SolutionData public latestSolution;
-
-    struct SolutionData {
-        uint32 batchId;
-        TradeData[] trades;
-        uint16[] tokenIdsForPrice;
-        address solutionSubmitter;
-        uint256 feeReward;
-        uint objectiveValue;
-    }
-
-    struct TradeData {
-        address owner;
-        uint128 volume;
-        uint16 orderId;
-    }
-
+    /** @dev a solver facing function called for auction settlement
+      * @param batchIndex index of auction solution is referring to
+      * @param owners array of addresses corresponding to touched orders
+      * @param orderIds array of order ids used in parallel with owners to identify touched order
+      * @param volumes executed buy amounts for each order identified by index of owner-orderId arrays
+      * @param prices list of prices for touched tokens indexed by next parameter
+      * @param tokenIdsForPrice price[i] is the price for the token with tokenID tokenIdsForPrice[i]
+      *
+      * Requirements:
+      * - Solutions for this `batchIndex` are currently being accepted.
+      * - feeToken price is specified.
```

Comment:
One minor and very unimportant thing in order to prove that I read it:
Upper case for feeToken

Reply:
What is upper case? `FeeToken`? `FEE_TOKEN`? `Fee Token`? I do not see anywhere in the contract that explicitly refers to this so I am unsure how to proceed with this one.

Reply:
I would just write FeeToken instead of feeToken, as everything starts with upper case letters. Otherwise, if you want to refer to the variable name, you might wanna use 'feeToken'.

But it is really minor

Reply:
```suggestion
      * - Fee Token price is non-zero
```